dd-java-agent/agent-tooling/src/test/groovy/datadog/trace/agent/decorator/BaseDecoratorTest.groovy

```diff
@@ -23,6 +24,7 @@ class BaseDecoratorTest extends DDSpecification {
     1 * span.setTag(DDTags.SPAN_TYPE, decorator.spanType())
     1 * span.setTag(Tags.COMPONENT.key, "test-component")
     _ * span.setTag(_, _) // Want to allow other calls from child implementations.
+    _ * span.getLocalRootSpan() // Want to allow other calls from child implementations.
     0 * _
   }
 
```

Comment:
One idea to better ensure all the paths that produce kind=server spans have been fixed is to go into TraceAssert.assertTrace and walk the spans in the trace, asserting that at most one has kind=server.  This would get applied to all the traces in the test suite.

Reply:
I had to add this "_ * span.getLocalRootSpan()" to allow ServerDecoratorTest to work, it actually not testing a new behavior. 

Most of the tests validate "span.kind" (that's why so many test files I have to update) but I really like your idea - if some instrumentation doesn't validate that, it is good place to catch it.

I've added an extra check, in TraceAssert.assertTrace as you suggested. Thanks.